enforcer/enforcer_main.py

```diff
@@ -0,0 +1,249 @@
+        was_mutated = len(patches) > 0
+        pod_admission_mutations.labels(mutated=str(was_mutated).lower()).inc()
+        admission_duration.labels(kind="Pod").observe(time.time() - start_time)
+
+        logger.debug(f"Pod patches %s", patches)
+
+        return {
+            "apiVersion": "admission.k8s.io/v1",
+            "kind": "AdmissionReview",
+            "response": {
+                "uid": request.request.get("uid"),
+                "allowed": True,
+                "patchType": "JSONPatch",
+                "patch": base64.b64encode(json.dumps(patches).encode()).decode() if patches else None
+            }
+        }
+        
+    except Exception as e:
+        logger.exception("Error processing webhook request")
+        raise HTTPException(status_code=500, detail=str(e))
+
+@app.get("/health")
+async def health_check():
+    """
+    Health check endpoint.
+    
+    Returns:
+        dict: Health status
+    """
+    return {"status": "healthy"}
+
+@app.get("/recommendations/{namespace}/{kind}/{name}")
+async def get_recommendations(namespace: str, kind: str, name: str):
+    """
+    Get recommendations for a workload.
+    
+    Args:
+        namespace: Kubernetes namespace
+        kind: Workload kind (e.g., Deployment, StatefulSet)
+        name: Workload name
+        
+    Returns:
+        dict: Recommendations per container or 404 if not found
+    """
+    try:
+        recommendations: WorkloadRecommendation = recommendation_store.get_recommendations(
+            name=name, namespace=namespace, kind=kind
+        )
+        
+        if not recommendations:
+            raise HTTPException(status_code=404, detail="No recommendations found for this workload")
+        
+        result = {}
+        for container_name, container_recommendation in recommendations.container_recommendations.items():
+            result[container_name] = {
+                "cpu": {
+                    "request": container_recommendation.cpu.request,
+                    "limit": container_recommendation.cpu.limit
+                } if container_recommendation.cpu else None,
+                "memory": {
+                    "request": container_recommendation.memory.request, 
+                    "limit": container_recommendation.memory.limit
+                } if container_recommendation.memory else None
+            }
+        
+        return {
+            "namespace": namespace,
+            "kind": kind, 
+            "name": name,
+            "containers": result
+        }
+        
+    except HTTPException:
+        raise
+    except Exception as e:
+        logger.exception("Error retrieving recommendations")
+        raise HTTPException(status_code=500, detail=str(e))
+
+@app.get("/metrics")
+async def metrics():
+    """
+    Prometheus metrics endpoint.
+    
+    Returns:
+        Response: Prometheus metrics in text format
+    """
+    # Update rs_owners size metric before returning metrics
```

Comment:
why?

Reply:
not updating it after the initial load from the api server
probably better to do it there. 
fixing it